### PR TITLE
issue with schema:load when a MySQL table definition contains partitions

### PR DESCRIPTION
**Pull request by ritikesh**
`bundle exec rails db:schema:load` fails with a SQL error when the table definition contains partitions.

### CODE REVIEW

**Comment by ritikesh** [comment]
CC: @kamipo - could this be a regression from #39365?

File: activerecord/lib/active_record/connection_adapters/mysql/schema_creation.rb [hunk 1]
```diff
@@ -45,9 +45,9 @@ def visit_IndexDefinition(o, create = false)
           end
 
           def add_table_options!(create_sql, o)
-            create_sql = super
             create_sql << " DEFAULT CHARSET=#{o.charset}" if o.charset
             create_sql << " COLLATE=#{o.collation}" if o.collation
+            super(create_sql, o)
             add_sql_comment!(create_sql, o.comment)
```
Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/mysql/schema_creation.rb:51]:
```suggestion
            add_sql_comment!(super, o.comment)
```

**Comment by kamipo** [comment]
> CC: @kamipo - could this be a regression from #39365?

Yes, this is a regression for #39365.

According to https://dev.mysql.com/doc/refman/8.0/en/create-table.html, *table_options* should be placed before *partition_options*.

File: activerecord/lib/active_record/connection_adapters/mysql/schema_creation.rb (hunk 1 above)
Comment (ritikesh) [reply activerecord/lib/active_record/connection_adapters/mysql/schema_creation.rb]: Done - https://github.com/rails/rails/pull/40631/commits/5b0850a019455614bc3283c2b24413dd515b2bb4

**Comment by kamipo** [comment]
Looks good to me. Can you squash your commits into one?

**Comment by ritikesh** [comment]
@kamipo, looks like I messed up with the git commands. 😅  I've reopened a new PR #40651.

**Pull request closed without merging**